src/amo/reducers/reviews.js

```diff
@@ -209,12 +189,50 @@ export const getReviewsByUserId = (
     : null;
 };
 
+export const latestByAddonVersionKey = ({
+  userId,
+  addonId,
+  versionId,
+}: {|
+  userId: number,
+  addonId: number,
+  versionId: number,
+|}) => {
+  return `user${userId}-addon${addonId}-version${versionId}`;
+};
+
+export const selectLatestUserReview = ({
+  reviewsState,
+  userId,
+  addonId,
+  versionId,
+}: {|
+  reviewsState: ReviewsState,
+  userId: number,
+  addonId: number,
+  versionId: number,
+|}): UserReviewType | null | void => {
+  const key = latestByAddonVersionKey({ userId, addonId, versionId });
+  const userReviewId = reviewsState.latestByAddonVersion[key];
+
+  if (userReviewId === null) {
+    // This means an action had previously attempted to fetch the latest
```

Comment:
Does that mean you make the distinction between `null` and `undefined` in the caller code?

Reply:
Otherwise, I would restrict the return value type to either `UserReviewType` or `null`.

Reply:
> Does that mean you make the distinction between null and undefined in the caller code?

Yes. The component only attempts to fetch a `userReview` [when the value is `undefined`](https://github.com/mozilla/addons-frontend/pull/5765/files#diff-883b82043ac0668a2dfdc54fe6914a8cR108). If the value is `null`, it knows that a fetch was already attempted and that no `userReview` exists.

This is a newer pattern that has been working really well, especially for SSR. This component was very old. It was looking for a user review every time an instance was created, regardless if it already had one or not. It was like smashing everything with hammer until it dies.